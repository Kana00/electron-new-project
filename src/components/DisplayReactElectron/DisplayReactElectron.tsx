import * as React from 'react';
import Lottie from 'react-lottie';
import Flexbox from 'flexbox-react';


export default class DisplayReactElectron extends React.Component<{}, {}> {
  render() {
    const styles = {
      viewContainer: {
        height: '70vh',
      },
      contentContainer: {
        marginTop: '1rem',
        height: '100%',
        weight: '100%',
      },
      text: {
        fontFamily: 'SourceCodeRoman-ExtraLight',
        color: '#BBBBBB',
        fontSize: 80,
        marginTop: -40,
      },
    }
    return (
      <Flexbox flexDirection='column' justifyContent='center' alignItems='center' style={styles.viewContainer}>
        <Flexbox flexDirection='column' justifyContent='center' alignItems='center' style={styles.contentContainer}>
          <Lottie
            options={{
              loop: false,
              autoplay: true,
              animationData: require('../../assets/animations/reactAnimation.json'),
            }}
            height={200}
            width={200}
            isStopped={false}
            isPaused={false}
          />
          <span style={styles.text}>React</span>
        </Flexbox>
      </Flexbox>
    );
  }
}
