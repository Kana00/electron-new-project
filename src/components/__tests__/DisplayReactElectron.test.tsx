import * as React from 'react';
import DisplayReactElectron from '../DisplayReactElectron/DisplayReactElectron';
import * as renderer from 'react-test-renderer';

jest.mock('react-lottie');

describe('Test component DisplayReactElectron', () => {
  it('render correctly', () => {
    const tree = renderer.create(<DisplayReactElectron />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
