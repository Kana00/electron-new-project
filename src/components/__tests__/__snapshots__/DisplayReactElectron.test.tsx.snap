// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component DisplayReactElectron render correctly 1`] = `
<div
  className="sc-bdVaJa gHzutc"
  style={
    Object {
      "height": "70vh",
    }
  }
>
  <div
    className="sc-bdVaJa gHzutc"
    style={
      Object {
        "height": "100%",
        "marginTop": "1rem",
        "weight": "100%",
      }
    }
  >
    <span
      style={
        Object {
          "color": "#BBBBBB",
          "fontSize": 80,
          "marginTop": -40,
        }
      }
    >
      React
    </span>
  </div>
</div>
`;
